model/execution/DeliveryExecutionManagerService.php

```diff
@@ -80,21 +81,43 @@ public function getDeliveryTimer($deliveryExecution)
      * Sets the extra time to a list of delivery executions
      * @param $deliveryExecutions
      * @param null $extraTime
+     * @param null $extendedTime
      * @return array
      */
-    public function setExtraTime($deliveryExecutions, $extraTime = null)
+    public function setExtraTime($deliveryExecutions, $extraTime = null, $extendedTime = null)
     {
         /** @var DeliveryMonitoringService $deliveryMonitoringService */
         $deliveryMonitoringService = $this->getServiceManager()->get(DeliveryMonitoringService::SERVICE_ID);
 
         $result = ['processed' => [], 'unprocessed' => []];
 
         /** @var DeliveryExecution $deliveryExecution */
-        foreach($deliveryExecutions as $deliveryExecution) {
+        foreach ($deliveryExecutions as $deliveryExecution) {
             if (is_string($deliveryExecution)) {
                 $deliveryExecution = $this->getDeliveryExecutionById($deliveryExecution);
             }
 
+            if ($extendedTime) {
+                /** @var TestSessionService $testSessionService */
+                $testSessionService = $this->getServiceLocator()->get(TestSessionService::SERVICE_ID);
+                $inputParameters = $testSessionService->getRuntimeInputParameters($deliveryExecution);
+                /** @var AssessmentTest $testDefinition */
+                $testDefinition = \taoQtiTest_helpers_Utils::getTestDefinition($inputParameters['QtiTestCompilation']);
+                $deliveryExecutionArray[] = $deliveryExecution;
+                $extraTime = null;
+                if ($maxTime = $testDefinition->getTimeLimits()->getMaxTime()) {
+                    $seconds = $maxTime->getSeconds(true);
+                    $secondsNew = $seconds * $extendedTime;
+                    $extraTime = floor(($secondsNew - $seconds) / 60) * 60;
+                    $deliveryMonitoringService = $this->getServiceLocator()->get(DeliveryMonitoringService::SERVICE_ID);
```

Comment:
`deliveryMonitoringService` is already in this scope, I think there is no reason to redefine it